File: Backend/Models/SensorModuleConfig.cs
namespace Solarponics.Models
{
    public class SensorModuleConfig
    {
        public string SerialNumber { get; set; }
        public string Name { get; set; }
        public string Location { get; set; }
        public string Room { get; set; }
        public SensorModuleSensorConfig SensorConfig { get; set; }
        public NetworkType NetworkType { get; set; }
        public WirelessConfig WirelessConfig { get; set; }
        public IpConfigType IpConfigType { get; set; }
        public IpConfig StaticIpConfig { get; set; }
        public string ServerAddress { get; set; }
    }
}

File: www/src/app/models/sensor-module-config.ts
import { SensorModuleSensorConfig } from "./sensor-module-sensor-config";
import { WirelessConfig } from "./wireless-config";
import { IpConfig } from "./ip-config";

export class SensorModuleConfig {
  serialNumber: string;
  name: string;
  location: string;
  room: string;
  sensorConfig: SensorModuleSensorConfig;
  networkType: string;
  wirelessConfig: WirelessConfig;
  ipConfigType: string;
  staticIpConfig: IpConfig;
  serverAddress: string;
}
